tests/test_unit/test_kinematics.py

```diff
@@ -182,3 +184,193 @@ def test_approximate_derivative_with_invalid_order(order):
+        # bboxes: missing keypoints dim
+        # e.g. comparing 2 individuals from the same data array
+        lambda position: (
+            position.isel(individuals=0),
+            position.isel(individuals=1),
+        ),
+        # individuals dim is 1D
+        # poses: multiple keypoints
+        # bboxes: missing keypoints dim
+        # e.g. comparing 2 single-individual data arrays
+        lambda position: (
+            position.where(
+                position.individuals == position.individuals[0], drop=True
+            ).squeeze(),
+            position.where(
+                position.individuals == position.individuals[1], drop=True
+            ).squeeze(),
+        ),
+        # both individuals and keypoints dims are scalar (poses only)
+        # e.g. comparing 2 individuals from the same data array,
+        # at the same keypoint
+        lambda position: (
+            position.isel(individuals=0, keypoints=0),
+            position.isel(individuals=1, keypoints=0),
+        ),
+        # individuals dim is scalar, keypoints dim is 1D (poses only)
+        # e.g. comparing 2 single-individual, single-keypoint data arrays
+        lambda position: (
+            position.where(
+                position.keypoints == position.keypoints[0], drop=True
+            ).isel(individuals=0),
+            position.where(
+                position.keypoints == position.keypoints[0], drop=True
+            ).isel(individuals=1),
+        ),
+    ],
+    ids=[
+        "dim_has_ndim_0",
+        "dim_has_ndim_1",
+        "core_dim_has_ndim_0",
+        "core_dim_has_ndim_1",
+    ],
+)
+def test_cdist_with_single_dim_inputs(valid_dataset, selection_fn, request):
+    """Test that the pairwise distances data array is successfully
+     returned regardless of whether the input DataArrays have
+    ``dim`` ("individuals") and ``core_dim`` ("keypoints")
+    being either scalar (ndim=0) or 1D (ndim=1),
+    or if ``core_dim`` is missing.
+    """
+    if request.node.callspec.id not in [
+        "core_dim_has_ndim_0-valid_bboxes_dataset",
+        "core_dim_has_ndim_1-valid_bboxes_dataset",
+    ]:  # Skip tests with keypoints dim for bboxes
+        valid_dataset = request.getfixturevalue(valid_dataset)
+        position = valid_dataset.position
+        a, b = selection_fn(position)
+        assert isinstance(kinematics.cdist(a, b, "individuals"), xr.DataArray)
+
+
+def expected_pairwise_distances(pairs, input_ds, dim):
+    """Return a list of the expected data variable names
+    for pairwise distances tests.
+    """
+    if pairs is None:
+        paired_elements = list(
+            itertools.combinations(getattr(input_ds, dim).values, 2)
+        )
+    else:
+        paired_elements = [
+            (elem1, elem2)
+            for elem1, elem2_list in pairs.items()
+            for elem2 in (
+                [elem2_list] if isinstance(elem2_list, str) else elem2_list
+            )
+        ]
```

Comment:
This bit of code just duplicates what's in the implementation, thus introducing some circularity in testing it?

Reply:
I've now removed this and spell out the expectations in parametrization